.cirrus.yml

```
@@ -0,0 +1,58 @@
+task:
+  trigger_type: manual
+  name: E2E/Integration Tests
+  timeout_in: 45
+  container:
+    image: ubuntu:20.04
+    kvm: true # required for E2E/Integrations tests
+    cpu: 6
+    memory: 12G
+  env:
+    DEBIAN_FRONTEND: noninteractive
+    CI: true
+    CIRRUS_CLONE_DEPTH: 0 # Required to get app version
+    FORCE_COLOR: 1 # force terminal color
+    PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1 # disable auto-browser download
+    KEYRING: /usr/share/keyrings/nodesource.gpg
+    NODE_VERSION: node_14.x # set node version e.g: "node_16.x" for Node 16 LTS
+
+  # Setting up a non-root user
+  prepare_user_script:
+    - groupadd --gid $(stat -c '%g' /dev/kvm) kvm
+    - useradd --create-home --groups kvm ranchertest
+    - tar c --owner=ranchertest . | tar x --directory=/home/ranchertest
+    - chown -R ranchertest:ranchertest /home/ranchertest
+
+  install_deps_script: |
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates gcc g++ curl git golang openssh-client make netcat sudo vim gpg lsb-core \
+                    libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+    export DISTRO="$(lsb_release --short --codename)"
+    curl --silent --location --fail --output /dev/null "https://deb.nodesource.com/${NODE_VERSION}/dists/${DISTRO}/Release"
+    curl --silent https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor > "$KEYRING"
```

```suggestion
    curl --silent https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
```
Hmm, seems to require installing `gpg-agent`… not sure it's worth it.

---

I don't think it worth to replace from `gpg` to `apt-key add`. It's also a recommended by nodesource using gqg, if we wanna to avoid the classic `sudo|bash` old school install.

Also it looks like there's no public key from nodesource:
```bash
curl --silent --location --fail --output /dev/null "https://deb.nodesource.com/${NODE_VERSION}/dists/${DISTRO}/Release"
curl --silent https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
Warning: apt-key output should not be parsed (stdout is not a terminal)
OK
echo "deb [signed-by=${KEYRING}] https://deb.nodesource.com/${NODE_VERSION} ${DISTRO} main" > /etc/apt/sources.list.d/nodesource.list
echo "deb-src [signed-by=${KEYRING}] https://deb.nodesource.com/${NODE_VERSION} ${DISTRO} main" >> /etc/apt/sources.list.d/nodesource.list
apt-get update 
Get:1 https://deb.nodesource.com/node_14.x focal InRelease [4583 B]
Get:2 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Err:1 https://deb.nodesource.com/node_14.x focal InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1655A0AB68576280
Hit:3 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:4 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:5 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Reading package lists...
W: GPG error: https://deb.nodesource.com/node_14.x focal InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1655A0AB68576280
E: The repository 'https://deb.nodesource.com/node_14.x focal InRelease' is not signed.
Exit status: 100
```